src/WidgetBase.php

```diff
@@ -191,7 +211,51 @@ public function setWeight($weight) {
   /**
    * {@inheritdoc}
    */
-  public function validate(array &$form, FormStateInterface $form_state) {}
+  public function prepareEntities(FormStateInterface $form_state) {
+    return [];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validate(array &$form, FormStateInterface $form_state) {
+    $trigger = $form_state->getTriggeringElement();
+
+    if (in_array('submit', $trigger['#array_parents'])) {
+      $entities = $this->prepareEntities($form_state);
+      $violations = $this->runWidgetValidators($entities);
+      if (!empty($violations)) {
+        /** @var \Symfony\Component\Validator\ConstraintViolationListInterface $violation */
+        foreach ($violations as $violation) {
+          $form_state->setError($form['widget']['upload'], $violation->getMessage());
+        }
+      }
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function runWidgetValidators(array $entities, $validators = []) {
+    // @todo Implement a centralized way to get arguments from path for all widgets?
+    if ($validators_hash = $this->request->get('validators')) {
+      $passed_validators = $this->keyValue->get($validators_hash);
+
+      if (!empty($passed_validators)) {
+        $validators += $passed_validators;
+      }
+    }
+
+    foreach ($validators as $validator_id => $options) {
+      /** @var \Drupal\entity_browser\WidgetValidationInterface $widget_validator */
+      $widget_validator = \Drupal::service('plugin.manager.entity_browser.widget_validation')->createInstance($validator_id, []);
+    }
+
+    if ($widget_validator) {
```

Comment:
We are only validating last validator. I think that this should be inside the loop.